images/ceph/Dockerfile

```diff
@@ -23,6 +23,7 @@ RUN curl --fail -sSL -o /tini https://github.com/krallin/tini/releases/download/
     chmod +x /tini
 
 COPY rook rookflex toolbox.sh /usr/local/bin/
+COPY ceph-csi /etc/ceph-csi
```

Comment:
This so the operator can call these templates directly in the image instead of asking the user to do it from the docs?

Reply:
Yes. FWIW I think its good if the user can override these templates but by default the operator should require as little from the user as possible.

Reply:
This seems fine for a temporary solution if it helps get the feature sooner, but with a plan to remove them soon. Could you add a comment here to this effect?

Reply:
What if we save this commit for another PR where the yaml files are consumed by the operator? By itself, it's hard to see the purpose of this change.

Reply:
Does this expect the `ceph-csi` binary to be part of the image? Where are we getting it from?

Reply:
If it's not clear from the diff the "ceph-csi" path in this case is a directory tree containing templates. The tree gets copied so that the image carries the default csi templates in order to avoid having the user attach them to the pod spec.

@travisn I'm good with the first comment and OK with the 2nd if the goal is to minimize the scope of this PR. However, I would say that the yaml files *are* consumed by the operator in that the csi deployment code requires that template files exist in these directories. The current approach requires the user to supply an alternative yaml file for setting up Rook in which the template files are attached to the pod spec via config maps. The user has to define the config maps ahead of time. This is documented in the "CSI Drivers Enablement" section of Rook's [Ceph CSI Drivers documentation](https://rook.io/docs/rook/master/ceph-csi-drivers.html) (on master). If you mean something else, could you please elaborate how you think the templates should be consumed?

Reply:
Ah, I see now that the manifests pulled into the image will be consumed since they're in the same path where the configmap was being mounted. But doesn't that mean we need to update the doc so they don't create the configmap anymore, and could we remove the env vars in operator-with-csi.yaml that specify the mount path?

Reply:
Yes to changing the docs and yes to cleaning up any env vars in the template that are no longer needed. Are you good with the general approach now? If so I'll start making corresponding changes to the docs as I had been holding back before we hashed out these more general design discussions.